plugin/trino-prometheus/src/main/java/io/trino/plugin/prometheus/PrometheusConnectorConfig.java

```diff
@@ -107,6 +110,33 @@ public PrometheusConnectorConfig setBearerTokenFile(File bearerTokenFile)
         return this;
     }
 
+    @NotNull
+    public Optional<String> getUser()
+    {
+        return Optional.ofNullable(user);
+    }
+
+    @Config("prometheus.auth.user")
+    public PrometheusConnectorConfig setUser(String user)
+    {
+        this.user = user;
+        return this;
+    }
+
+    @NotNull
+    public Optional<String> getPassword()
+    {
+        return Optional.ofNullable(password);
+    }
+
+    @Config("prometheus.auth.password")
+    @ConfigSecuritySensitive
+    public PrometheusConnectorConfig setPassword(String password)
```

Comment:
Can we add some validation like if the userName is configured then we require the password and vice versa ?

Reply:
Is it possible to have both the basic and token based authentication ?

Reply:
Not sure if Prometheus supports two authentications. Added guard logic to allow only single auth.